src/main/java/com/github/manolo8/darkbot/gui/utils/inspector/InspectorTree.java

```diff
@@ -116,26 +143,151 @@ private void editValue(boolean primitivesOnly) {
             copyValueItem.addActionListener(a -> {
                 ObjectTreeNode node = getSelectedNode();
-                if (node != null) SystemUtils.toClipboard(node.strValue);
+                if (node != null) SystemUtils.toClipboard(node.getText());
             });
             copyAddressItem.addActionListener(a -> {
                 ObjectTreeNode node = getSelectedNode();
-                if (node != null) SystemUtils.toClipboard(String.format("0x%x", node.address.get()));
+                if (node != null) SystemUtils.toClipboard(String.format("0x%x", node.address.getAsLong()));
+            });
+
+            copyClosureAddressItem.addActionListener(a -> {
+                ObjectTreeNode node = getSelectedNode();
+                if (node != null) {
+                    ObjectInspector.Slot slot = node.getSlot();
+                    if (slot.slotType == ObjectInspector.Slot.Type.OBJECT && !slot.name.contains("$")) {
+                        SystemUtils.toClipboard(String.format("0x%x", ByteUtils.getClassClosure(node.value)));
+                    }
+                }
             });
+
+            copyClassStructureItem.addActionListener(this::generateClass);
             editValueItem.addActionListener(a -> editValue(false));
+            editTypeNameItem.addActionListener(a -> {
+                ObjectTreeNode node = getSelectedNode();
+                if (node != null) {
+                    ObjectInspector.Slot slot = node.getSlot();
+                    String result = JOptionPane.showInputDialog(getRootPane(),
+                            "Edit type name of " + slot.type + " " + slot.name, "Edit type name", JOptionPane.PLAIN_MESSAGE);
+                    if (result != null && !result.isEmpty())
+                        slot.setReplacement(result);
+                }
+            });
+
+            copyValueItem.setAccelerator(KeyStroke.getKeyStroke(KeyEvent.VK_V, 0));
+            copyAddressItem.setAccelerator(KeyStroke.getKeyStroke(KeyEvent.VK_A, 0));
+            copyClosureAddressItem.setAccelerator(KeyStroke.getKeyStroke(KeyEvent.VK_C, 0));
+            copyClassStructureItem.setAccelerator(KeyStroke.getKeyStroke(KeyEvent.VK_S, 0));
+            editValueItem.setAccelerator(KeyStroke.getKeyStroke(KeyEvent.VK_E, 0));
+
             add(copyValueItem);
             add(copyAddressItem);
+            add(copyClosureAddressItem);
+            add(copyClassStructureItem);
             add(editValueItem);
+            add(editTypeNameItem);
+        }
+
+        private void generateClass(ActionEvent e) {
+            ObjectTreeNode node = getSelectedNode();
+            if (node != null) {
+                ObjectInspector.Slot slot = node.getSlot();
+                if (slot.slotType == ObjectInspector.Slot.Type.OBJECT || slot.slotType == ObjectInspector.Slot.Type.PLAIN_OBJECT) {
+
+                    StringBuilder clazzBuilder = new StringBuilder("import com.github.manolo8.darkbot.core.itf.Updatable;");
```

Comment:
this thing def needs to go in a diff class, and consider making it a template, as in, a resource text-file with some stuff like {className}, {methods} or w/e that you can search-and-replace instead of having the whole thing written with appends